test/uk/gov/hmrc/digitalservicestaxfrontend/connectors/DSTConnectorTest.scala

```diff
@@ -51,6 +54,22 @@ class DSTConnectorTest extends WiremockSpec with ConfiguredPropertyChecks {
     }
   }
 
+  "should have correct status code where registration failed" in {
+    val dstRegNumber = arbitrary[Registration].sample.value
+
+    stubFor(
+      post(urlPathEqualTo("/registration"))
+        .willReturn(aResponse().withStatus(Status.BAD_REQUEST).withBody("{}")))
+
+    val submitReg =
```

Comment:
Just a query.. I am not sure what is the purpose of converting the request submitReg to a Future[Boolean] . Can the code simple be: 
      val submitReturn =
        DSTTestConnector.submitReturn(period, ret)

      whenReady(submitReturn) {threwException => assert(threwException.status == 400)}

Reply:
I guess I was originally thinking a property based test would throw a range of exceptions. I can change if you think this is a show stopper.